### PR TITLE
feat: define list accessor for bigframes Series

### PR DESCRIPTION
**Pull request by sycai**

### CODE REVIEW

File: bigframes/series.py [hunk 1]
```diff
@@ -161,6 +162,10 @@ def query_job(self) -> Optional[bigquery.QueryJob]:
     def struct(self) -> structs.StructAccessor:
         return structs.StructAccessor(self._block)
 
+    @property
+    def list(self) -> lists.ListAccessor:
```
Comment (chelsea-lin) [inline bigframes/series.py:166]: please add the doctest on https://github.com/googleapis/python-bigquery-dataframes/blob/main/third_party/bigframes_vendored/pandas/core/series.py

Comment (chelsea-lin) [reply bigframes/series.py:166]: Also, please add the new accessor to docs/reference/bigframes.pandas/series.rst

**Review by chelsea-lin — CHANGES_REQUESTED** [review]

File: bigframes/series.py (hunk 1 above)
Comment (sycai) [reply bigframes/series.py:166]: Change updated. PTAL!

File: third_party/bigframes_vendored/pandas/core/arrays/arrow/accessors.py [hunk 2]
```diff
@@ -6,6 +6,79 @@
 from bigframes import constants
 
 
+class ListAccessor:
+    """
+    Accessor object for list data properties of the Series values.
+    """
+
+    def len(self):
+        """
```
Comment (chelsea-lin) [inline third_party/bigframes_vendored/pandas/core/arrays/arrow/accessors.py:15]: It looks like the doc format could be aligned with Google's style. Could you take another look at the formatting in `StringMethod.str()`?

File: third_party/bigframes_vendored/pandas/core/arrays/arrow/accessors.py [hunk 3]
```diff
@@ -6,6 +6,79 @@
...
+        Return the length of each list in the Series.
+
+        Returns
+        -------
+        Series
+            The length of each list.
+
+        See Also
+        --------
+        str.len : Python built-in function returning the length of an object.
```
Comment (chelsea-lin) [inline third_party/bigframes_vendored/pandas/core/arrays/arrow/accessors.py:25]: Also, please review the `See also` format in `Series.case_when`. Of the listed APIs, `StringMethods.len` seems most relevant to our current implementation. Maybe we can just keep this one and remove other two?

File: third_party/bigframes_vendored/pandas/core/arrays/arrow/accessors.py (hunk 2 above)
Comment (sycai) [reply third_party/bigframes_vendored/pandas/core/arrays/arrow/accessors.py]: Good call. I was not aware that the formats are different. Updated.

File: third_party/bigframes_vendored/pandas/core/arrays/arrow/accessors.py (hunk 3 above)
Comment (sycai) [reply third_party/bigframes_vendored/pandas/core/arrays/arrow/accessors.py]: Done

**Review by chelsea-lin — APPROVED** [review]
Thanks!

**Pull request merged**